app/components/primer/beta/flash_banner.ts

```diff
@@ -0,0 +1,62 @@
+import {controller, target} from '@github/catalyst'
+
+@controller
+class FlashBannerElement extends HTMLElement {
+  @target root: HTMLElement
+  @target message: HTMLElement
+
+  private lastFocusedElement: HTMLElement
+
+  connectedCallback() {
+    this.lastFocusedElement = (document.activeElement as HTMLElement) || document.body
+
+    if (document.readyState.toString() === 'ready' || document.readyState.toString() === 'complete') {
+      this.announceMessage()
+    } else {
+      document.addEventListener('readystatechange', () => {
+        if (document.readyState.toString() === 'complete') {
+          this.announceMessage()
+        }
+      })
+    }
+  }
+
+  dismiss() {
+    if (this.shouldReappear()) {
+      this.style.setProperty('visibility', 'hidden')
+
+      setTimeout(() => {
+        this.style.setProperty('visibility', 'visible')
+      }, 2000)
+
+      return
+    }
+
+    const parentElement = this.parentElement
+    if (!parentElement) return
+
+    parentElement.removeChild(this)
+    this.lastFocusedElement?.focus()
+  }
+
+  private announceMessage() {
+    setTimeout(() => {
+      this.message.focus()
+    }, 200)
```

Comment:
Is there concern that focusing on delay can interrupt a user who is already starting to interact with the page? Something I jotted down in [my coaching findings was](https://github.com/github/accessibility/pull/2101/files#diff-5f93c27804a1cf56989f041af5356de22a12434616f5a05508873c22b57f5817R60-R62):

> In testing, I found that if we have a `role=”alert”` element and update the content with a non-breaking space on `load` state with a 200ms delay, we can pretty consistently get screen reader announcements. It is possible that focus management on load & delay alone will lead to SR announcement, but by the time page load completes, user may already be interacting with the page so we should not

Reply:
Hmm yeah that's a valid concern. Maybe we can just remove the delay then?

Reply:
> Maybe we can just remove the delay then?

Yeah, I think that removing the delay for focus events will minimize potential issues.

> Adding a non-breaking space to the content didn't work when I was testing in Lookbook 😢

It's possible the delay wasn't long enough given waiting on the `load` state in dotcom will result in a longer wait. Also if you use ` ` instead of `nbsp;`, the SR announcement will not kick off. Let me know if there's a branch I can check out!

Reply:
I ended up increasing the delay to 500ms and it worked in all browsers with VoiceOver and NVDA 🎉